lumicks/pylake/fdcurve.py

```diff
@@ -79,6 +80,31 @@ def __getitem__(self, item):
         new_curve.stop = new_curve.f._src.stop
         return new_curve
 
+    def _slice_by_distance(self, min_dist, max_dist, max_gap=3):
+        """Return a subset of this FD curve within a selected distance range"""
+        # logical mask of data lying within selected distance range
+        mask_in_range = np.logical_and(min_dist <= self.d.data, self.d.data <= max_dist).astype(np.bool)
+
+        # find runs of data that are not in selected distance range
+        # runs shorter than the allowed gap should be allowed
+        ranges, lengths = find_contiguous(~mask_in_range)
+        for rng, ln in zip(ranges, lengths):
+            if ln < max_gap:
+                # ignore first and last runs of data, as total length is not defined
+                if rng[0] == 0 or rng[1] == self.d.data.size:
```

Comment:
I'm not sure I'd understand from the comment what this condition is trying to catch.

Maybe something like 
`# We want to include short sections of data that fall outside the selected distance range, when these excursions are caused by noise. However, we do not want to extend to the start or end of the curve, since we cannot know whether we are simply going out of the range because of noise, or whether we have just actually left the range.`